.github/workflows/ci-wheels.yml

```diff
@@ -72,7 +72,7 @@ jobs:
         UDUNITS2_XML_PATH: ${{ matrix.xml_path }}
         CIBW_ENVIRONMENT_PASS_LINUX: UDUNITS2_INCDIR UDUNITS2_LIBDIR UDUNITS2_XML_PATH
 
-    - uses: actions/upload-artifact@v3
+    - uses: actions/upload-artifact@v4
       with:
         name: pypi-artifacts
```

Comment:
As we;ve seen the basic changes is that artifacts are now[ immutable and non-rewritable](https://github.com/actions/toolkit/tree/main/packages/artifact#breaking-changes)
 
So I think the problem here is probably that the artifact name must be different for each usage, 
so, something like `pypi-artifacts-${{ matrix.os }}-${{ matrix.arch }}` might work ?

Reply:
Yeah, there is now no merging and named have to be unique